### DISCUSSION TITLE
How to use an isolated state file per project

### DISCUSSION THREAD

**Discussion by siepkes** [scaleway/terraform-provider-scaleway#2614] [Q&A]
Hi all!

I would like to use an (isolated) separate state file per project. Mainly in order to have development, staging and production projects (i.e. environments) which can't mess with each other. Is this possible? Setting the `project_id` in the provider does not seem to stop Terraform from only working on resources in that specific project. It still tries to delete the resources in other projects it does not know about.

**Comment by remyleone** [comment]
Have you tried the recent terraform stacks? https://developer.hashicorp.com/terraform/language/stacks this feature I think could be helpful to implement what you describe. In case you are in tofu there is some support with https://opentofu.org/docs/language/state/workspaces/

**Comment by siepkes** [comment]
Thanks for the suggestion! Though we have migrated to OpenTofu which does not have this feature. 

Also it seems you can only use stacks with Hashicorp's commercial Terraform offering, HCP.
